log4j-core/src/main/java/org/apache/logging/log4j/core/config/builder/api/ConfigurationBuilder.java

```diff
@@ -61,6 +61,15 @@ public interface ConfigurationBuilder<T extends Configuration> extends Builder<T
      */
     ConfigurationBuilder<T> add(CustomLevelComponentBuilder builder);
 
+    /**
+     * Adds a top level component.
+     * @param builder The ComponentBuilder with all of its attributes and sub components set.
+     * @return this builder instance.
+     */
+    default ConfigurationBuilder<T> addComponent(ComponentBuilder<?> builder) {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
Can you add a `@since` tag?

Instead of adding a default method that throws an exception, I would prefer introducing an abstract method and annotating the class with [@ProviderType](https://docs.osgi.org/javadoc/osgi.annotation/7.0.0/org/osgi/annotation/versioning/ProviderType.html).

From the perspective of consumers of `ConfigurationBuilder`, the behavior remains effectively the same: if an implementation does not define the method, it will fail at runtime (via either `AbstractMethodError` or `UnsupportedOperationException`). However, for implementors of `ConfigurationBuilder`, declaring it as an abstract method is more explicit and has the advantage of failing at compile time, prompting them to implement the method directly.

Reply:
Added the `@since` tag in 2fea3657a6507069d997ecaefc94e8e013d314d5.

> I would prefer introducing an abstract method

This is an `interface`, not an `abstract class`, do you imply not providing any default methods at all?

Reply:
Yes, that would be my preferred option.

If someone implemented `ConfigurationBuilder`, he can expect compilation errors after a minor version bump.

Reply:
It causes `bnd-baseline:baseline` to fail, i.e., that'd be a backward incompatible change.